controllers/storagecluster/reconcile.go

```diff
@@ -496,6 +497,38 @@ func (r *StorageClusterReconciler) reconcilePhases(
 
 			// mark operator upgradeable if and only if all storageclusters are ready
 			if r.clusters.AreOtherStorageClustersReady(instance) {
+				if instance.Spec.AllowRemoteStorageConsumers {
+					scList := &ocsv1alpha1.StorageConsumerList{}
+					err := r.Client.List(r.ctx, scList, &client.ListOptions{Namespace: instance.Namespace})
+					if err != nil {
+						r.Log.Error(err, "Failed to list StorageConsumers")
+						return reconcile.Result{}, err
+					}
+					providerVersion, err := semver.Make(version.Version)
+					if err != nil {
+						r.Log.Error(err, "Failed to parse OCS Version")
+						return reconcile.Result{}, err
+					}
+					for idx := range scList.Items {
+						clientVersion, err := semver.Make(scList.Items[idx].Status.Client.OperatorVersion)
+						if err != nil {
+							r.Log.Error(err, "Failed to parse Client operator version")
+							return reconcile.Result{}, err
+						}
+						// onboarded client operator can be either on same provider operator version or one less in minor version
+						// we need to stop provider operator upgrades for these lagging client operators only
+						// because if provider operator upgrades, that lagging client operator will become unsupported
+						if providerVersion.Minor-1 == clientVersion.Minor {
```

Comment:
To cover all cases can we just do:
 ` clientVersion.Major < providerVersion.Major || clientVersion.Minor < providerVersion.Minor` to have a stronger check and capture any edge cases

Reply:
- if we are depending on onboarding check for same versioned client operators we can forgo Major version check
- for Minor version, will change to `lt` from current `eq` match.
---
- if we are not depending on onboarding check, neither existing check nor the suggested one will work, like a client operator might already be ahead of provider operator

Reply:
done.